src/registries/menus/number_format_menu_registry.ts

```diff
@@ -56,11 +59,64 @@ numberFormatMenuRegistry
   })
   .add("more_formats", {
     ...ACTION_FORMAT.moreFormats,
-    sequence: 110,
+    sequence: 120,
   });
 
+export function getCustomNumberFormats(
+  env: SpreadsheetChildEnv
+): ACTION_FORMAT.NumberFormatActionSpec[] {
+  const defaultFormats = new Set(
+    numberFormatMenuRegistry
+      .getAll()
+      .map((f) => (typeof f.format === "function" ? f.format(env) : f.format))
+  );
+
+  const customFormats = new Map<Format, ACTION_FORMAT.NumberFormatActionSpec>();
+  for (const sheetId of env.model.getters.getSheetIds()) {
+    const cells = env.model.getters.getEvaluatedCells(sheetId);
```

Comment:
It's worth checking the performance on large spreadsheet. It won't be a nice experience if the menu takes 100+ms to open

Reply:
I tested a bit on a big spreadsheet, and it feels fine (<100ms).

Actually in my first implementation I stored the custom formats in a plugin that reacted to commands, but then every update cell command became slower 😅

Reply:
I guess we can't have the beurre et l'argent du beurre...

Reply:
on rng spreadsheet, the first hit is 200+ms but it still feels very usable  👍